icu4j/main/core/src/main/java/com/ibm/icu/text/UnicodeSet.java

```diff
@@ -5127,5 +5143,179 @@ public static void setDefaultXSymbolTable(XSymbolTable xSymbolTable) {
+     * Returns a {@link Stream} of {@link EntryRange} values from this {@code UnicodeSet}.
+     *
+     * <p><b>Warning:</b> The {@link EntryRange} instance is the same each time; the contents are just reset.
+     *
+     * <p><b>Warning:</b> To iterate over the full contents, you have to also iterate over the strings.
+     *
+     * <p><b>Warning:</b> For speed, {@code UnicodeSet} iteration does not check for concurrent modification.<br>
+     * Do not alter the {@code UnicodeSet} while iterating.
+     *
+     * @return a {@link Stream} of {@link EntryRange}
+     */
+    public Stream<EntryRange> rangeStream() {
+        return StreamSupport.stream(ranges().spliterator(), false);
+    }
+
+    /**
+     * Returns a {@link Stream} of {@code String} values from this {@code UnicodeSet}.
+     *
+     * <p><b>Warning:</b> To iterate over the full contents, you have to also iterate over the strings.
+     *
+     * <p><b>Warning:</b> For speed, {@code UnicodeSet} iteration does not check for concurrent modification.<br>
+     * Do not alter the {@code UnicodeSet} while iterating.
+     *
+     * @return a {@link Stream} of {@code String}
+     */
+    public Stream<String> stringStream() {
+        return StreamSupport.stream(strings().spliterator(), false);
+    }
+
+    /**
+     * Returns an {@link IntStream} of Unicode code point values from this {@code UnicodeSet}.
+     *
+     * <p><b>Warning:</b> To iterate over the full contents, you have to also iterate over the strings.
+     *
+     * <p><b>Warning:</b> For speed, {@code UnicodeSet} iteration does not check for concurrent modification.<br>
+     * Do not alter the {@code UnicodeSet} while iterating.
+     *
+     * @return an {@link IntStream} of Unicode code point values
+     */
+    public IntStream codePointStream() {
+        return StreamSupport.intStream(new CodePointsSpliterator(this), false);
+    }
+
+    /**
+     * Returns a stream of {@code String} values from this {@code UnicodeSet}.
+     *
+     * <p><b>Warning:</b> To iterate over the full contents, you have to also iterate over the strings.
+     *
+     * <p><b>Warning:</b> For speed, {@code UnicodeSet} iteration does not check for concurrent modification.<br>
+     * Do not alter the {@code UnicodeSet} while iterating.
+     *
+     * @return a {@link Stream} of {@code String}
+     */
+    public Stream<String> stream() {
+        return StreamSupport.stream(spliterator(), false);
+    }
+
+    /**
+     * Provide for iteration on code points.
+     *
+     * <p>Returns an {@link Iterable} over all the code points, "expanding" the ranges
+     * by iterating on all code points between the {@link EntryRange#codepoint} and
+     * {@link EntryRange#codepointEnd}.
+     *
+     * <p><b>Warning:</b> This is a convenience method, but comes with a performance penalty
+     * because it boxes {@code int} into {@code Integer}.<br>
+     * For faster iteration use the {@link #codePointStream()} or {@link #charAt(int)}.
```

Comment:
charAt(index) is not fast -- it linearly walks through each range until it finds the one with the index-th code point. Each time.
Better to point to UnicodeSetIterator.next() for an efficient but old alternative.